### PR TITLE
Should fix achievement spam bug.

### PR DESCRIPTION
**Pull request by Raycoms**
Review please @Minecolonies/maintainers

and tweak speed slightly.

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BuildingHome.java#L166](https://github.com/Minecolonies/minecolonies/blob/39a12b629ff885c85279ffa60fde8139e3f0fa5d/src/main/java/com/minecolonies/colony/buildings/BuildingHome.java#L166): Remove this unused "owner" local variable. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1481)
1. MAJOR [BuildingHome.java#L166](https://github.com/Minecolonies/minecolonies/blob/39a12b629ff885c85279ffa60fde8139e3f0fa5d/src/main/java/com/minecolonies/colony/buildings/BuildingHome.java#L166): Remove this useless assignment to local variable "owner". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)

**Comment by Kostronor** [comment]
:shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BuildingHome.java#L166](https://github.com/Minecolonies/minecolonies/blob/69fe05b6021a9c0919e6a352b751d028d10571af/src/main/java/com/minecolonies/colony/buildings/BuildingHome.java#L166): Remove this unused "owner" local variable. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1481)
1. MAJOR [BuildingHome.java#L166](https://github.com/Minecolonies/minecolonies/blob/69fe05b6021a9c0919e6a352b751d028d10571af/src/main/java/com/minecolonies/colony/buildings/BuildingHome.java#L166): Remove this useless assignment to local variable "owner". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)

File: src/main/java/com/minecolonies/colony/buildings/BuildingHome.java [hunk 1]
```diff
@@ -167,11 +167,11 @@ public void onUpgradeComplete(final int newLevel)
 
         if (newLevel == 1)
         {
-            owner.triggerAchievement(ModAchievements.achievementBuildingColonist);
+            this.getColony().triggerAchievement(ModAchievements.achievementBuildingColonist);
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/colony/buildings/BuildingHome.java]: Is this the only place we have missed this?

Comment (cltnschlosser) [inline src/main/java/com/minecolonies/colony/buildings/BuildingHome.java]: :shipit: doesn't look like it.

**Pull request merged**